engines/bastion_katello/app/assets/javascripts/bastion_katello/products/discovery/discovery-create.controller.js

```diff
@@ -172,15 +172,9 @@ angular.module('Bastion.products').controller('DiscoveryCreateController',
             return fieldsEnabled;
         };
 
-        $scope.gpgKeys = function () {
-            var deferred = $q.defer();
-
-            GPGKey.queryUnpaged(function (response) {
-                deferred.resolve(response.results);
-            });
-
-            return deferred.promise;
-        };
+        GPGKey.queryUnpaged(function (gpgKeys) {
+            $scope.gpgKeys = gpgKeys.results;
+        });
```

Comment:
Does this need to handle error conditions?  Currently an error (404, etc.) would fail silently.